### PR TITLE
Add styles for authorization process

### PR DESCRIPTION
**Pull request by RayRedGoose**
### This PR contains the next changes:
- Added welcome page styles
- Added login page styles
- Added reset password page styles
- Added change password page styles
- Added signup page styles
- Added verification page styles

Also, I added 2 new routes:
- `/auth/:type` for authorization pages with the same design
- `/verification` for verification process pages

I didn't touch any functionality, but I modified some components structure for design realization.
Also, I didn't change the error notifications and breadcrumbs for now.

### CODE REVIEW

File: src/views/AuthorizationPage.vue [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        <LogIn />
+      </div>
+      <div v-if="$route.params.type==='reset'">
+        <h2>Forgot password?</h2>
+        <ResetPsw />
+      </div>
+      <div v-if="$route.params.type==='new'">
+        <h2>Create new password</h2>
+        <NewPsw />
+      </div>
```
Comment (wlonk) [inline src/views/AuthorizationPage.vue:19]: I would prefer this as 4 separate components, each wired in to the router distinctly.

File: src/components/SignUp.vue [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          type="password"
+          id="rep-password"
+          placeholder="Password"
+          class="ui input"
+        />
+      </div>
+    </div>
+    <div class="check-row row">
+      <label for="checkbox" class="checkbox-label">
+        <input type="checkbox" id="checkbox" />
```
Comment (wlonk) [inline src/components/SignUp.vue:35]: For some reason I can't tab to this checkbox, and when I click it, it's very hard to see that it's checked.

File: src/components/SignUp.vue [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        </router-link>
+      </p>
+    </div>
+  </form>
+</template>
+
+<script>
+export default {
+  name: 'SignUp'
+};
```
Comment (wlonk) [inline src/components/SignUp.vue:56]: You mentioned you hadn't done functionality yet; would it be helpful if I got you the specs on the registration API endpoint?

File: src/components/NewPsw.vue [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    </div>
+    <div class="submit-row row">
+      <input type="submit" value="Reset password" class="submit-btn" />
+    </div>
+  </form>
+</template>
+
+<script>
+export default {
+  name: 'NewPsw'
```
Comment (wlonk) [inline src/components/NewPsw.vue:31]: I would prefer longer but more explicit names; `ResetPasswordForm` would be ideal.

File: src/components/ResetPsw.vue [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+          Log In
+        </router-link>
+      </p>
+    </div>
+  </form>
+</template>
+
+<script>
+export default {
+  name: 'ResetPsw'
```
Comment (wlonk) [inline src/components/ResetPsw.vue:20]: Here too: `RequestResetLinkForm`.

File: src/components/LogIn.vue [hunk 6]
```diff
@@ -29,16 +32,32 @@
...
-        :key="i"
-        class="ui error message"
-      >
-        {{ error }}
-      </li>
-    </ul>
+    <div class="check-row row">
+      <label for="checkbox" class="checkbox-label">
+        <input type="checkbox" id="checkbox" checked="true"/>
+        <p>Remember me</p>
```
Comment (wlonk) [inline src/components/LogIn.vue:38]: Right now, the frontend automatically remembers; I'm inclined to leave this off until I make functionality to allow it to forget!

File: src/components/LogIn.vue [hunk 7]
```diff
@@ -1,7 +1,8 @@
 <template>
-  <form @submit.stop.prevent="login" class="ui mini form">
-    <div class="ui input">
-      <input name="username" placeholder="username" />
+  <form @submit.stop.prevent="login" class="page-form">
```
Comment (wlonk) [inline src/components/LogIn.vue:2]: We'll need to add a "navigate to `/`" line to the login handler, eventually.

File: src/views/RootPage.vue [hunk 8]
```diff
@@ -1,8 +1,15 @@
...
     <DashBoard v-if="currentUser" />
-    <div v-else>
-      <h1>Welcome to Roughcast</h1>
+    <div v-else class="welcome-block">
+      <div>
+        <h1>Welcome to Roughcast</h1>
+        <p>Roughcast is a tool to help tabletop game designers disseminate playtest documents, and manage conversations around those documents.</p>
+        <router-link to="/auth/signup" class="submit-btn">
+          Get Started!
+        </router-link>
```
Comment (wlonk) [inline src/views/RootPage.vue:10]: I'd love a little rounded-corner on the `submit-btn` class; it feels just a bit stark and sharp as it stands.

File: src/components/SignUp.vue (hunk 2 above)
Comment (RayRedGoose) [reply src/components/SignUp.vue:35]: fixed.

File: src/components/LogIn.vue (hunk 6 above)
Comment (RayRedGoose) [reply src/components/LogIn.vue:38]: I can comment it out until you add this functionality

File: src/views/RootPage.vue (hunk 8 above)
Comment (RayRedGoose) [reply src/views/RootPage.vue:10]: I added rounded corners to all inputs, so now it connected with other styles elements.

File: src/components/LogIn.vue (hunk 6 above)
Comment (RayRedGoose) [reply src/components/LogIn.vue:38]: or I can make it as a read-only

File: src/views/AuthorizationPage.vue (hunk 1 above)
Comment (RayRedGoose) [reply src/views/AuthorizationPage.vue]: ok, I fixed it.

File: src/components/ResetPsw.vue (hunk 5 above)
Comment (RayRedGoose) [reply src/components/ResetPsw.vue]: done for both.

**Comment by RayRedGoose** [comment]
Finally, I've made other changes based on your comments.

**Comment by RayRedGoose** [comment]
@wlonk In addition, I recreated user panel in a header based on design wireframes and update notification dropdown menu styles.

File: src/components/LogIn.vue (hunk 6 above)
Comment (wlonk) [reply src/components/LogIn.vue:42]: Oh, I rather like that as a way of communicating that it's always the case.

File: src/components/UserPanel.vue [hunk 9]
```diff
@@ -0,0 +1,22 @@
+<template>
+  <div class="user-panel">
+    <router-link :to="`/u/${currentUser.username}`" class="ui link item">
+      <img src="../assets/no-avatar.svg" alt="avatar">
```
Comment (wlonk) [inline src/components/UserPanel.vue:4]: Is this path based on this file's location at compile-time, and does it get set to a compiled asset absolute location? I can play around with it and answer my own question.

Comment (RayRedGoose) [reply src/components/UserPanel.vue:4]: I used the router link that you've created for redirection to the user page.

**Comment by wlonk** [comment]
OK, final question: do you wanna do the functionality or do you mind if I wire that in quick, before merging?

**Comment by RayRedGoose** [comment]
> OK, final question: do you wanna do the functionality or do you mind if I wire that in quick, before merging?

You can do it. I've started to built layouts with styles for the user profile page.

**Pull request merged**